internal/overlord/checkstate/checkers.go

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) 2021 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package checkstate
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"os/exec"
+	"strconv"
+	"syscall"
+
+	"github.com/canonical/pebble/internal/logger"
+	"github.com/canonical/pebble/internal/osutil"
+	"github.com/canonical/pebble/internal/strutil/shlex"
+)
+
+// httpChecker is a checker that ensures an HTTP GET at a specified URL returns 20x.
+type httpChecker struct {
+	name    string
+	url     string
+	headers map[string]string
+}
+
+func (c *httpChecker) check(ctx context.Context) error {
```

Comment:
do these actually belong in a package called "checkstate"

Reply:
I see what you mean. They could go in their own internal package and the checkers could be exported. However, I think the pragmatic placement in this package is okay -- they're very much related, and it's likely the only place they'll ever be used. If that changes, we can always move them to a separate package.